### PR TITLE
Upgrade log4j to version 2.

### PR DESCRIPTION
**Pull request by oleksandr-yatsenko1**
Fix security vulnerability reported in log4j 1.2 (EDGRTAC-26)
https://issues.folio.org/browse/EDGRTAC-26

### CODE REVIEW

**Comment by mrdpopov** [comment]
How did you make sure the vulnerability is fixed?
I thought the solution was to update to **log4j2**, but in this case it's still 1.7.30
You can find examples in mod-oai-pmh, mod-inventory-storage
Please also add configuration for log4j2 likes it's done in mod-oai-pmh, for example

```
                  <name>log4j2.configuration</name>
                  <value>${project.baseUri}src/main/resources/log4j2.properties</value>

...
           <vertx.logger-delegate-factory-class-name>io.vertx.core.logging.Log4j2LogDelegateFactory</vertx.logger-delegate-factory-class-name>

```

**Comment by oleksandr-yatsenko1** [comment]
> How did you make sure the vulnerability is fixed?
> I thought the solution was to update to **log4j2**
> You can find examples in mod-oai-pmh, mod-inventory-storage
> Please also add configuration for log4j2 likes it's done in mod-oai-pmh, for example
> 
> ```
>                   <name>log4j2.configuration</name>
>                   <value>${project.baseUri}src/main/resources/log4j2.properties</value>
> 
> ...
>            <vertx.logger-delegate-factory-class-name>io.vertx.core.logging.Log4j2LogDelegateFactory</vertx.logger-delegate-factory-class-name>
> ```

Thnak toy for recommendations, I'll use the same approach here.

File: pom.xml [hunk 1]
```diff
@@ -109,27 +109,40 @@
...
+      <type>pom</type>
     </dependency>
 
     <!-- include slf4j bridges to proxy all logging from library deps to
       our logging impl -->
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>jcl-over-slf4j</artifactId>
-      <version>1.7.18</version>
+      <version>1.7.30</version>
```
Comment (craigmcnally) [inline pom.xml:145]: If this is always going to move at the same pace as slf4j-api, you might consider pulling this out into a variable so it only needs to be changed in one place.

File: pom.xml [hunk 2]
```diff
@@ -109,27 +109,40 @@
...
+      The Log4j 2 SLF4J Binding allows applications coded to the SLF4J API to use Log4j 2 as the implementation.
+
+      Due to a break in compatibility in the SLF4J binding, as of release 2.11.1 two SLF4J to Log4j Adapters are provided.
+
+      log4j-slf4j-impl should be used with SLF4J 1.7.x releases or older.
+      log4j-slf4j18-impl should be used with SLF4J 1.8.x releases or newer.
+      Applications that take advantage of the Java Module System should use SLF4J 1.8.x and log4j-slf4j18-impl.
+
+      The Log4j 2 SLF4J Binding has a dependency on the Log4j 2 API as well as the SLF4J API
+    -->
```
Comment (craigmcnally) [inline pom.xml:125]: I appreciate the comments - this will probably be helpful when we move to newer versions of the jdk.

Comment (oleksandr-yatsenko1) [reply pom.xml:125]: Sure.

File: pom.xml (hunk 1 above)
Comment (oleksandr-yatsenko1) [reply pom.xml:145]: Good point, will fix.

**Pull request merged**